ion/marshal.go

```diff
@@ -447,7 +446,7 @@ func (m *Encoder) encodeTimeDate(v reflect.Value) error {
 	ns := t.Nanosecond()
 	numFractionalSeconds := 0
 	if ns > 0 {
-		numFractionalSeconds = len(strconv.Itoa(ns))
+		numFractionalSeconds = 9
```

Comment:
I wonder if this shouldn't instead be a constant rather than an embedded magic number. Could you also add a comment describing what's going on here?

Reply:
Updated to use a const and added a comment